sound/soc/sof/loader.c

```diff
@@ -185,9 +188,12 @@ int snd_sof_run_firmware(struct snd_sof_dev *sdev)
 	}
 
 	if (sdev->ipc->ops->post_fw_boot)
-		return sdev->ipc->ops->post_fw_boot(sdev);
+		ret = sdev->ipc->ops->post_fw_boot(sdev);
 
-	return 0;
+	dev_dbg(sdev->dev, "firmware boot (post) complete, took %lldus\n",
+		ktime_us_delta(ktime_get(), boot_start));
```

Comment:
I think this is pointless to print out if the ops->post_fw_boot is not set, I would move it inside of the if() and print it only if `ret == 0` with "Full firmware startup took %lldus\n".
Or you might want to break it down to firmware boot proper and post boot time.

Reply:
I still think that this is pointless to print if `sdev->ipc->ops->post_fw_boot == NULL`

Also, a theoretical log:
```
firmware boot complete, took 100us
firmware boot (post) complete, took 140us
```
the full boot time is 100us+140us or 140us? I know the answer, but from the print I assume the former, while the later is the meaning.

and w/o post_fw_boot:
```
firmware boot complete, took 100us
firmware boot (post) complete, took 101us
```
The printk took 1us, is that really something which helps?